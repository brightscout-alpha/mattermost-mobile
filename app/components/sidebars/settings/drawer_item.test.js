// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallowWithIntl} from 'test/intl-test-helper';

import Preferences from '@mm-redux/constants/preferences';

import DrawerItem from './drawer_item';
import {Text} from 'react-native';

describe('DrawerItem', () => {
    const baseProps = {
        onPress: jest.fn(),
        testID: 'test-id',
        centered: true,
        defaultMessage: 'default message',
        i18nId: 'i18-id',
        iconName: 'icon-name',
        isDestructor: true,
        separator: true,
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <DrawerItem {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with label sibling element and centered and separator false', () => {
        const labelSibling = (
            <Text>{'Hey'}</Text>
        );
        const wrapper = shallowWithIntl(
            <DrawerItem
                {...baseProps}
                centered={false}
                separator={false}
                labelSibling={labelSibling}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
