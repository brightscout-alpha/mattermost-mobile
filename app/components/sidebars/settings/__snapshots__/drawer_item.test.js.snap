// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerItem should match snapshot 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  testID="test-id"
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "padding": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "marginLeft": 5,
          "width": 45,
        }
      }
    >
      <CompassIcon
        name="icon-name"
        style={
          Array [
            Object {
              "color": "rgba(61,60,64,0.64)",
              "fontSize": 24,
            },
            Object {
              "color": "#fd5960",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "width": "70%",
          }
        }
      >
        <FormattedText
          defaultMessage="default message"
          id="i18-id"
          style={
            Array [
              Object {
                "color": "rgba(61,60,64,0.5)",
                "flex": 1,
                "fontSize": 17,
                "includeFontPadding": false,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#fd5960",
              },
              Object {
                "textAlign": "center",
                "textAlignVertical": "center",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(61,60,64,0.2)",
            "height": 1,
          }
        }
      />
    </View>
  </View>
</ForwardRef>
`;

exports[`DrawerItem should match snapshot with label sibling element and centered and separator false 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  testID="test-id"
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "padding": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "marginLeft": 5,
          "width": 45,
        }
      }
    >
      <CompassIcon
        name="icon-name"
        style={
          Array [
            Object {
              "color": "rgba(61,60,64,0.64)",
              "fontSize": 24,
            },
            Object {
              "color": "#fd5960",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "width": "70%",
          }
        }
      >
        <FormattedText
          defaultMessage="default message"
          id="i18-id"
          style={
            Array [
              Object {
                "color": "rgba(61,60,64,0.5)",
                "flex": 1,
                "fontSize": 17,
                "includeFontPadding": false,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#fd5960",
              },
              Object {},
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 14,
            "top": 3,
          }
        }
      >
        <Text>
          Hey
        </Text>
      </View>
    </View>
  </View>
</ForwardRef>
`;
